### PR TITLE
Bump bot core version

### PR DESCRIPTION
**Pull request by ChrisLovering**
This also bumps all the other deps while I'm here, and updated the module name.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
 python = "3.10.*"
 
 # See https://bot-core.pythondiscord.com/ for docs.
-bot-core = { url = "https://github.com/python-discord/bot-core/archive/refs/tags/v8.2.1.zip", extras = ["async-rediscache"] }
-redis = "4.3.4"
-fakeredis = { version = "1.9.3", extras = ["lua"] }
+pydis_core = { version = "9.1.1", extras = ["async-rediscache"] }
+redis = "4.3.5"
+fakeredis = { version = "2.0.0", extras = ["lua"] }
```
Comment (ChrisLovering) [inline pyproject.toml:14]: This removes support for aioredis, since we use redis-py we're fine.

File: pyproject.toml [hunk 2]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
...
 
 aiohttp = "3.8.3"
 arrow = "1.2.3"
 beautifulsoup4 = "4.11.1"
-colorama = { version = "0.4.5", markers = "sys_platform == 'win32'" }
+colorama = { version = "0.4.6", markers = "sys_platform == 'win32'" }
 coloredlogs = "15.0.1"
-deepdiff = "5.8.1"
-emoji = "2.1.0"
+deepdiff = "6.2.1"
```
Comment (ChrisLovering) [inline pyproject.toml:21]: > Note: There are no breaking changes in DeepDiff 6 compared to the latest DeepDiff 5 releases.

From https://github.com/seperman/deepdiff#what-is-new

File: pyproject.toml [hunk 3]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
...
+emoji = "2.2.0"
 feedparser = "6.0.10"
 lxml = "4.9.1"
 
 # Must be kept on this version unless doc command output is fixed
 # See https://github.com/python-discord/bot/pull/2156
 markdownify = "0.6.1"
 
-more-itertools = "8.14.0"
+more-itertools = "9.0.0"
```
Comment (ChrisLovering) [inline pyproject.toml:30]: Breaking changes are only to functions we do not use.
https://more-itertools.readthedocs.io/en/stable/versions.html

File: pyproject.toml [hunk 4]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
...
+rapidfuzz = "2.13.2"
+regex = "2022.10.31"
+sentry-sdk = "1.11.1"
 tldextract = "3.4.0"
 pydantic = "1.10.2"
 
 [tool.poetry.dev-dependencies]
 coverage = "6.5.0"
-flake8 = "5.0.4"
+flake8 = "6.0.0"
```
Comment (ChrisLovering) [inline pyproject.toml:42]: 6.0.0 removes the --diff option, that pre-coomit doesn't use.

https://flake8.pycqa.org/en/latest/release-notes/6.0.0.html#backwards-incompatible-changes

File: pyproject.toml [hunk 5]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
...
 flake8-string-format = "0.3.0"
 flake8-tidy-imports = "4.8.0"
 flake8-todo = "0.7"
-flake8-isort = "5.0.0"
+flake8-isort = "5.0.3"
 pep8-naming = "0.13.2"
 pre-commit = "2.20.0"
-pip-licenses = "3.5.4"
-pytest = "7.1.3"
+pip-licenses = "4.0.1"
```
Comment (ChrisLovering) [inline pyproject.toml:52]: Dropped support for 3.7 and internal changes that we don't rely on.
https://github.com/raimon49/pip-licenses/releases/tag/v-4.0.0

File: pyproject.toml [hunk 6]
```diff
@@ -9,53 +9,52 @@ license = "MIT"
...
+pytest = "7.2.0"
 pytest-cov = "4.0.0"
 python-dotenv = "0.21.0"
-pytest-xdist = "2.5.0"
-pytest-subtests = "0.8.0"
+pytest-xdist = "3.0.2"
```
Comment (ChrisLovering) [inline pyproject.toml:56]: Removes the `--boxed` cli arg, which we don't use.

**Review by mbaruh — APPROVED** [review]

**Review by HassanAbouelela — APPROVED** [review]

**Pull request merged**